src/ripple/app/ledger/impl/LedgerToJson.cpp

```diff
@@ -99,46 +99,42 @@ void fillJsonBinary(Object& json, bool closed, LedgerInfo const& info)
     }
 }
 
-Json::Value fillJsonTx (LedgerFill const& fill,
-    bool bBinary, bool bExpanded,
-        std::pair<std::shared_ptr<STTx const>,
-            std::shared_ptr<STObject const>> const i)
+Json::Value
+fillJsonTx(
+    LedgerFill const& fill,
+    bool bBinary,
+    bool bExpanded,
+    std::shared_ptr<STTx const> const& txn,
+    std::shared_ptr<STObject const> const& stMeta)
 {
     if (! bExpanded)
     {
-        return to_string(i.first->getTransactionID());
+        return to_string(txn->getTransactionID());
     }
     else
     {
         Json::Value txJson{ Json::objectValue };
-        auto const txnType=i.first->getTxnType();
+        auto const txnType=txn->getTxnType();
         if (bBinary)
         {
-            txJson[jss::tx_blob] = serializeHex(*i.first);
-            if (i.second)
-            {
-                if (txnType == ttPAYMENT || txnType == ttCHECK_CASH)
-                {
-                    // TBD Insert delivered_amount somehow
-                }
-                txJson[jss::meta] = serializeHex(*i.second);
-            }
+            txJson[jss::tx_blob] = serializeHex(*txn);
+            if (stMeta)
+                txJson[jss::meta] = serializeHex(*stMeta);
         }
         else
         {
-            copyFrom(txJson, i.first->getJson(0));
-            if (i.second)
+            copyFrom(txJson, txn->getJson(0));
+            if (stMeta)
             {
-                txJson[jss::metaData] = i.second->getJson(0);
+                txJson[jss::metaData] = stMeta->getJson(0);
                 if (txnType == ttPAYMENT || txnType == ttCHECK_CASH)
                 {
-                    auto const& txn = i.first;
+                    // Insert delivered amount
                     auto const& txnID = txn->getTransactionID();
-                    auto rawMeta = fill.ledger.txRead(txnID).second;
-                    if (rawMeta)
+                    if (stMeta)
```

Comment:
This check is unnecessary, `stMeta` must be valid here.

Reply:
Agreed. Fixed in `Mickey fixes 2` (to be pushed)